OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -220,14 +220,15 @@ void IBotTick.BotTick(IBot bot)
 			// PERF: We tick only one type of valid queue at a time
 			// if AI gets enough cash, it can fill all of its queues with enough ticks
 			var findQueue = false;
+			var queuesByCategory = AIUtils.FindQueuesByCategory(player);
 			for (int i = 0, builderIndex = currentBuilderIndex; i < builders.Length; i++)
 			{
 				if (++builderIndex >= builders.Length)
 					builderIndex = 0;
 
 				--builders[builderIndex].WaitTicks;
 
-				var queues = AIUtils.FindQueues(player, builders[builderIndex].Category).ToArray();
+				var queues = queuesByCategory[builders[builderIndex].Category].ToArray();
```

Comment:
The `ToArray` doesn't have much added value anymore? More cost than gain?

Reply:
Without doing this, it's an `IEnumerable`, and we want to be able to check the length and then loop over it later on. So keeping `ToArray` is easiest for that.